torch/distributed/rendezvous.py

```diff
@@ -88,6 +93,33 @@ def rendezvous(url: str, rank: int = -1, world_size: int = -1, **kwargs):
         raise RuntimeError("No rendezvous handler for {}://".format(result.scheme))
     return _rendezvous_handlers[result.scheme](url, **kwargs)
 
+def _create_store_from_options(backend_options, rank):
+    result = urlparse(backend_options.init_method)
+
+    # If using env initialization, get rank and world_size from env
+    world_size = -1
+    if result.scheme == "env":
+        rank = os.environ.get("RANK", rank)
+        # world_size default set to -1
+        # if the world_size env variable is not present then it is a dynamic group
```

Comment:
Could you please write a complete explanation of what condition will trigger dynamic group?

Reply:
Added additional explanation in `__init__.py` The complete explanation is just that if `world_size` is not set in `init_rpc` and also not set in environment variable `WORLD_SIZE`, then that triggers dynamic group initialization